src/test_application/universes/vehicles.cpp

```diff
@@ -223,38 +259,39 @@ osp::universe::Satellite testapp::debug_add_part_vehicle(
 
     auto& engBP = blueprint.add_part(engine, cfOset + feOset, idRot, scl);
 
-    Quaternion yz090 = rotZ_090 * rotY_090;
-    Quaternion yz180 = rotZ_180 * rotY_090;
-    Quaternion yz270 = rotZ_270 * rotY_090;
+    // Add a shit ton of RCS rings
+
+    Rad rcsRingStep     = 90.0_degf;
+    float rcsZMin       = -2.0f;
+    float rcsZMax       = 2.1f;
+    float rcsZStep      = 4.0f;
+    float rcsRadius     = 1.1f;
 
-    // Top RCS ring
-    blueprint.add_part(rcs, rcsOsetTop, rotY_090, scl);
-    blueprint.add_part(rcs, rotZ_090.transformVector(rcsOsetTop), yz090, scl);
-    blueprint.add_part(rcs, rotZ_180.transformVector(rcsOsetTop), yz180, scl);
-    blueprint.add_part(rcs, rotZ_270.transformVector(rcsOsetTop), yz270, scl);
+    std::vector<int> rcsPorts;
 
-    // Top RCS ring
-    blueprint.add_part(rcs, rcsOsetBtm, rotY_090, scl);
-    blueprint.add_part(rcs, rotZ_090.transformVector(rcsOsetBtm), yz090, scl);
-    blueprint.add_part(rcs, rotZ_180.transformVector(rcsOsetBtm), yz180, scl);
-    blueprint.add_part(rcs, rotZ_270.transformVector(rcsOsetBtm), yz270, scl);
+    for (float z = rcsZMin; z < rcsZMax; z += rcsZStep)
+    {
+        Vector3 rcsOset = cfOset + Vector3{rcsRadius, 0.0f, z};
+
+        // Add RCS rings from 0 to 360 degrees
+        for (Rad ang = 0.0_degf; ang < Rad(360.0_degf); ang += rcsRingStep)
+        {
+            Quaternion rotZ = Quaternion::rotation(ang, Vector3{0, 0, 1});
+            blueprint_add_rcs_block(blueprint, rcs, rcsPorts,
+                                    rotZ.transformVector(rcsOset),
+                                    rotZ * rotY_090);
+        }
+    }
 
     enum Parts
     {
         CAPSULE = 0,
         FUSELAGE = 1,
-        ENGINE = 2,
-        RCS1 = 3,
```

Comment:
Recommend making this an enum class, BTW.

Reply:
Enum class is sort of annoying for this because these are supposed to just be named indices, so you'd need `static_cast<int>()` everywhere you want to use the enum. I think it's fine as-is since it's just local to this function scope.